### DISCUSSION TITLE
Generating FIT files when Garmin auth isn't working

### DISCUSSION THREAD

**Discussion by douglasr** [philosowaffle/peloton-to-garmin#538] [General]
Perhaps it's common knowledge and/or documented somewhere, but I spent an hour poking around the docs and couldn't figure out how to generate a FIT file that I could manually upload to Garmin Connect during any authentication outages.

Step 1 (which was documented) was to turn on the "Save a backup copy of Converted files to output dir" setting in Conversion -> Advanced. However, because of the current auth issues with Garmin Connect, FIT files were still not being generated. I suspect that the code is set up such that authentication happens first, and then the requested workouts are converted and uploaded. However, because the authentication flow with Garmin fails, the entire process, including the conversion stops (@philosowaffle can perhaps confirm).

Step 2, which solved my problem, was to go to the Garmin settings and turn off "Automatically upload to Garmin Connect". After changing this setting, you simply run the sync process manually like you would if Garmin was working. The process runs successfully, creates the backup files in the output directory and reports "Successfully synced selected workouts" even though, of course, nothing was actually synced.

**Comment by philosowaffle** [comment]
Do you have any logs you can share for when you were trying to do only Step 1? From peeking at the code real quick it should be working as the files are saved prior to trying to authenticate with Garmin.

**Reply by douglasr** [reply]
I didn't have the "Save a backup" option turned on prior to this, and I also use MFA with Garmin, so it might be related to either of those things? However, there is no "Converting..." entry in the logs when I first ran it. I've attached the logs.
[log20230930.txt](https://github.com/philosowaffle/peloton-to-garmin/files/12777749/log20230930.txt)

**Reply by philosowaffle** [reply]
Ah yes this makes sense now, this seems specific to the MFA flow on the UI. In non-MFA, P2G will download and convert first, then auth with Garmin. In the MFA flow, P2G attempts to get a Garmin auth token first since it requires user input for the code.  I need to update the docs to reflect this difference in behavior.
